tests/unit/ai/review/test_orchestrator.py

```
@@ -198,12 +198,108 @@ def _recording_call_ai(
 
+def test_run_review_partial_when_cost_cap_before_any_chunk() -> None:
+    """Cap tripping before any chunk completes returns an actionable partial.
+
+    A depth-2 chunk overspends the cap on its question-generation call, so the
+    main review budget check raises before the chunk produces a partial. The
+    result must be a clean, empty partial (``partial=True``, zero chunks
+    reviewed) rather than the generic abort error.
+    """
+    provider = _mock_provider(content=_sample_response_json())
+
+    def _recording_call_ai(*, provider, budget=None, **kwargs):  # noqa: ANN001, ANN202
```

Ruff (0.15.20)

[warning] Missing type annotation for `**kwargs`

(ANN003)